staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -182,6 +196,92 @@ func (p *Patcher) buildMergePatch(original, modified, current []byte) ([]byte, e
 	return patch, nil
 }
 
+// gvkSupportsPatchOpenAPIV3 checks if a particular GVK supports the patch operation.
+// It returns an error if the OpenAPI V3 could not be downloaded.
+func (p *Patcher) gvkSupportsPatchOpenAPIV3(gvk schema.GroupVersionKind) (bool, error) {
+	// Bypassing root to save apiserver memory.
+	gvSpec, err := p.OpenAPIV3Root.GVSpec(schema.GroupVersion{
+		Group:   p.Mapping.GroupVersionKind.Group,
+		Version: p.Mapping.GroupVersionKind.Version,
+	})
+	if err != nil {
+		return false, err
+	}
+	// TODO: Better error checking
+	for _, path := range gvSpec.Paths.Paths {
+		if path.Patch != nil {
+			if gvkMatches(p.Mapping.GroupVersionKind, path.Patch.Extensions) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// TODO: GVK parser should be moved elsewhere
+// Is this duplicated somewhere?
+func gvkMatches(targetGVK schema.GroupVersionKind, ext spec.Extensions) bool {
+	if ext == nil {
+		return false
+	}
+	f, ok := ext["x-kubernetes-group-version-kind"]
```

Comment:
Is there a constant somewhere for this string (`x-kubernetes-group-version-kind`)?

Reply:
There is in `util/openapi` in kubectl but that's coupled with OpenAPI V2 and I was hoping we can keep OpenAPI V2 and V3 coupling to a minimum. I declared a constant in this file instead.